main.go

```diff
@@ -410,35 +423,102 @@ func (b *Backend) updateCallStats(t shortTraceMsg) {
 }
 
 type multisite struct {
-	sites []*site
+	sites          []*site
+	healthCanceler context.CancelFunc
+	rwLocker       sync.RWMutex
 }
 
-func (m *multisite) populate(cellText [][]string) {
+func (m *multisite) renewSite(ctx *cli.Context, healthCheckPath string, healthReadCheckPath string, healthCheckPort int, healthCheckDuration, healthCheckTimeout time.Duration) {
+	ctxt, cancel := context.WithCancel(context.Background())
+	var sites []*site
+	for i, siteStrs := range ctx.Args() {
+		if i == len(ctx.Args())-1 {
+			healthCheckPath = healthReadCheckPath
+		}
+		site := configureSite(ctxt, ctx, i+1, strings.Split(siteStrs, ","), healthCheckPath, healthCheckPort, healthCheckDuration, healthCheckTimeout)
+		sites = append(sites, site)
+	}
+	m.rwLocker.Lock()
+	defer m.rwLocker.Unlock()
+	m.sites = sites
```

Comment:
You don't need locking at all if you're using atomic operations. It's too complicated to show here, so I pushed it to this commit https://github.com/ramondeklein/sidekick/commit/e2be194cb8c4d087ceaa7a4c3207870602e88298.